.github/workflows/build-package.yml

```diff
@@ -3,84 +3,79 @@ name: Build Package
-        type: string
-        required: false
-        default: '[{"arch": "arm", "platform": "linux/arm/v7"}, {"arch": "arm64", "platform": "linux/arm64/v8"}, {"arch": "amd64", "platform": "linux/amd64"}]'
       artifact:
-        description: The artifact to publish.
+        description: The artifact output containing the packages from this workflow.
         type: string
         required: false
-        default: 'package'
+        default: "package"
       build-type:
-        description: The build type to use.
+        description: The type of build to perform (Debug | Release).
         type: string
-        default: Release
-        required: false
-      signed:
-        description: If the packages should be signed. This will create a second artifact for the signed packages (<artifact_name>-signed).
-        type: boolean
         required: false
-        default: true
+        default: "Release"
+      os:
+        description: The operating system to build for.
+        type: string
+        required: true
+      arch:
+        description: The architecture to build for.
+        type: string
+        required: true
+      platform:
+        description: The platform to build for.
+        type: string
+        required: true
+
+env:
+  BUILD_TYPE: Release
+  MOUNT: /azure-osconfig
+  REGISTRY: ghcr.io
 
 jobs:
   package:
     name: Package
     runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        os: ${{ fromJson(inputs.os) }}
-        variant: ${{ fromJson(inputs.variant) }}
-    env:
-      container-workspace: /azure-osconfig
+
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: recursive
           clean: true
 
-      - name: Set version
-        id: version
-        run: echo "tweak=$(echo $(date +'%Y%m%d%H%M'))" >> $GITHUB_OUTPUT
-
       - name: Run container
         id: container
         uses: ./.github/actions/container-run
         with:
-          username: ${{ secrets.ACR_CLIENT_ID }}
-          password: ${{ secrets.ACR_CLIENT_SECRET }}
-          os: ${{ matrix.os }}
-          arch: ${{ matrix.variant.arch }}
-          platform: ${{ matrix.variant.platform }}
-          mount: ${{ github.workspace }}:${{ env.container-workspace }}
+          registry: ${{ env.REGISTRY }}
+          container: azure/azure-osconfig/${{ inputs.os }}-${{ inputs.arch }}
+          mount: ${{ github.workspace }}:${{ env.MOUNT }}
+          platform: ${{ inputs.platform }}
 
       - name: Generate build
         uses: ./.github/actions/container-exec
         with:
           container: ${{ steps.container.outputs.id }}
           cmd: |
+            # Set the tweak version (YYMMDD + 2 digit run number + 2 digit run attempt)
+            tweak=$(date +'%Y%m%d')
+            tweak="${tweak}$(printf '%02d' ${{ github.run_number }})"
+            tweak="${tweak}$(printf '%02d' ${{ github.run_attempt }})"
```

Comment:
can we just have the `run_number`, we shouldn't care for the workflow attempt

Reply:
The reason for differentiating between attempts is to prevent collisions if a package job is re-run. However, I agree, it makes sense to only use `run_number` since these jobs will likely not ever be re-run without some additional changes to the package.